Mathlib/Analysis/NormedSpace/Alternating/Basic.lean

```diff
@@ -0,0 +1,601 @@
+
+nonrec theorem le_mul_prod_of_opNorm_le_of_le
+    {m : ι → E} {C : ℝ} {b : ι → ℝ} (hC : ‖f‖ ≤ C) (hm : ∀ i, ‖m i‖ ≤ b i) :
+    ‖f m‖ ≤ C * ∏ i, b i :=
+  f.le_mul_prod_of_opNorm_le_of_le hC hm
+
+theorem le_opNorm_mul_prod_of_le (f : E [⋀^ι]→L[𝕜] F) {b : ι → ℝ} (hm : ∀ i, ‖m i‖ ≤ b i) :
+    ‖f m‖ ≤ ‖f‖ * ∏ i, b i :=
+  f.1.le_opNorm_mul_prod_of_le hm
+
+nonrec theorem le_opNorm_mul_pow_card_of_le (f : E [⋀^ι]→L[𝕜] F) {m b} (hm : ‖m‖ ≤ b) :
+    ‖f m‖ ≤ ‖f‖ * b ^ Fintype.card ι :=
+  f.le_opNorm_mul_pow_card_of_le hm
+
+nonrec theorem le_opNorm_mul_pow_of_le {n} (f : E [⋀^Fin n]→L[𝕜] F) {m b} (hm : ‖m‖ ≤ b) :
+    ‖f m‖ ≤ ‖f‖ * b ^ n :=
+  f.le_opNorm_mul_pow_of_le hm
+
+theorem le_of_opNorm_le {C : ℝ} (h : ‖f‖ ≤ C) (m : ι → E) : ‖f m‖ ≤ C * ∏ i, ‖m i‖ :=
+  f.1.le_of_opNorm_le h m
+
+theorem ratio_le_opNorm (f : E [⋀^ι]→L[𝕜] F) (m : ι → E) : ‖f m‖ / ∏ i, ‖m i‖ ≤ ‖f‖ :=
+  f.1.ratio_le_opNorm m
+
+/-- The image of the unit ball under a continuous alternating map is bounded. -/
+theorem unit_le_opNorm (f : E [⋀^ι]→L[𝕜] F) (h : ‖m‖ ≤ 1) : ‖f m‖ ≤ ‖f‖ := f.1.unit_le_opNorm h
+
+/-- If one controls the norm of every `f x`, then one controls the norm of `f`. -/
+theorem opNorm_le_bound (f : E [⋀^ι]→L[𝕜] F) {M : ℝ} (hMp : 0 ≤ M)
+    (hM : ∀ m, ‖f m‖ ≤ M * ∏ i, ‖m i‖) : ‖f‖ ≤ M :=
+  f.1.opNorm_le_bound hMp hM
+
+theorem opNorm_le_iff {C : ℝ} (hC : 0 ≤ C) : ‖f‖ ≤ C ↔ ∀ m, ‖f m‖ ≤ C * ∏ i, ‖m i‖ :=
+  f.1.opNorm_le_iff hC
+
+/-- The fundamental property of the operator norm of a continuous alternating map:
+`‖f m‖` is bounded by `‖f‖` times the product of the `‖m i‖`, `nnnorm` version. -/
+theorem le_opNNNorm (f : E [⋀^ι]→L[𝕜] F) (m : ι → E) : ‖f m‖₊ ≤ ‖f‖₊ * ∏ i, ‖m i‖₊ :=
+  f.1.le_opNNNorm m
+
+theorem le_of_opNNNorm_le {C : ℝ≥0} (h : ‖f‖₊ ≤ C) (m : ι → E) : ‖f m‖₊ ≤ C * ∏ i, ‖m i‖₊ :=
+  f.1.le_of_opNNNorm_le h m
+
+theorem opNNNorm_le_iff {C : ℝ≥0} : ‖f‖₊ ≤ C ↔ ∀ m, ‖f m‖₊ ≤ C * ∏ i, ‖m i‖₊ :=
+  f.1.opNNNorm_le_iff
+
+theorem isLeast_opNNNorm (f : E [⋀^ι]→L[𝕜] F) :
+    IsLeast {C : ℝ≥0 | ∀ m, ‖f m‖₊ ≤ C * ∏ i, ‖m i‖₊} ‖f‖₊ :=
+  f.1.isLeast_opNNNorm
+
+theorem opNNNorm_prod (f : E [⋀^ι]→L[𝕜] F) (g : E [⋀^ι]→L[𝕜] G) :
+    ‖f.prod g‖₊ = max (‖f‖₊) (‖g‖₊) :=
+  f.1.opNNNorm_prod g.1
+
+theorem opNorm_prod (f : E [⋀^ι]→L[𝕜] F) (g : E [⋀^ι]→L[𝕜] G) : ‖f.prod g‖ = max (‖f‖) (‖g‖) :=
+  f.1.opNorm_prod g.1
+
+theorem opNNNorm_pi {ι' : Type*} [Fintype ι'] {F : ι' → Type*} [∀ i', SeminormedAddCommGroup (F i')]
+    [∀ i', NormedSpace 𝕜 (F i')] (f : ∀ i', E [⋀^ι]→L[𝕜] F i') : ‖pi f‖₊ = ‖f‖₊ :=
+  ContinuousMultilinearMap.opNNNorm_pi fun i ↦ (f i).1
+
+theorem opNorm_pi {ι' : Type*} [Fintype ι'] {F : ι' → Type*} [∀ i', SeminormedAddCommGroup (F i')]
+    [∀ i', NormedSpace 𝕜 (F i')] (f : ∀ i', E [⋀^ι]→L[𝕜] F i') : ‖pi f‖ = ‖f‖ :=
+  ContinuousMultilinearMap.opNorm_pi fun i ↦ (f i).1
+
+instance instNormedSpace {𝕜' : Type*} [NormedField 𝕜'] [NormedSpace 𝕜' F] [SMulCommClass 𝕜 𝕜' F] :
```

Comment:
Can we also have the `BoundedSMul` version of this, or would it still require `NormedField`?